### PR TITLE
API module initialization

### PR DESCRIPTION
**Pull request by hkulekci**
- Core routes moved to Frontend Module
 - Core Views moved to Frontend Module
 - Api UserResource created and zf-hal, zf-rest, zf-content-negotiation, zf-api-problem modules added to application modules for API module
 - zf-mvc-auth module added to composer json for api oauth2 implementations

### CODE REVIEW

**Comment by edigu** [comment]
:+1:

**Comment by hkulekci** [comment]
@edigu we can share a postman collection in `docs` or `data` folder? What do you think?

**Comment by edigu** [comment]
I think writing documentation in markdown format  with few request examples is enough.

File: composer.json [hunk 1]
```diff
@@ -16,7 +16,13 @@
...
         "gedmo/doctrine-extensions": "2.4.7",
         "mathiasverraes/money": "1.3.*@dev",
-        "zendframework/zend-i18n-resources": "^2.5"
+        "zendframework/zend-i18n-resources": "^2.5",
+        "zfcampus/zf-api-problem": "~1.0",
+        "zfcampus/zf-content-negotiation": "~1.0",
+        "zfcampus/zf-content-validation": "~1.0",
+        "zfcampus/zf-hal": "~1.0",
+        "zfcampus/zf-mvc-auth": "~1.0",
+        "zfcampus/zf-rest": "~1.0"
```
Comment (hkulekci) [inline composer.json]: @hkulekci Following dependencies coming from `zf-rest`

```
        "zfcampus/zf-api-problem": "~1.0",
        "zfcampus/zf-content-negotiation": "~1.0",
        "zfcampus/zf-hal": "~1.1",
        "zfcampus/zf-mvc-auth": "~1.1"
```

Comment (edigu) [inline composer.json]: Sure. :+1:

**Pull request merged**